src/algorithms/calorimetry/ImagingPixelReco.h

```diff
@@ -130,18 +124,19 @@ class ImagingPixelReco {
                         pos.x() / m_lUnit, pos.y() / m_lUnit, pos.z() / m_lUnit
                 );
 
-                hits.push_back(new edm4eic::CalorimeterHit{id,                         // cellID
-                                                           static_cast<float>(energy), // energy
-                                                           0,                          // energyError
-                                                           static_cast<float>(time),   // time
-                                                           0,                          // timeError TODO
-                                                           position,                   // global pos
-                                                           {0, 0, 0}, // @TODO: add dimension
-                                                           sid, lid,
-                                                           local});                    // local pos
-            }catch(std::exception &e){
+                recohits->create(id,                         // cellID
+                            static_cast<float>(energy), // energy
+                            0,                          // energyError
+                            static_cast<float>(time),   // time
+                            0,                          // timeError TODO
+                            position,                   // global pos
+                            edm4hep::Vector3f({0, 0, 0}), // @TODO: add dimension
```

Comment:
Can we avoid temporary with just an initializer list?

Reply:
That doesn't work. An std::forward is used and it doesn't seem to allow inference to figure out that a constructor should be called.